demisto_sdk/commands/common/hook_validations/field_base_validator.py

```diff
@@ -362,3 +365,62 @@ def does_not_have_empty_select_values(self) -> bool:
                                  warning=self.structure_validator.quite_bc):
                 return False
         return True
+
+    def is_aliased_fields_are_valid(self) -> bool:
+        """
+        Validates that the aliased fileds are valid.
+        invalid aliased filed are
+        1. field with marketplaces other than [`xsoar`]
+        2. filed with alias (aliased field with alias)
+
+        Returns:
+            (bool): True if aliased fields are valid.
+        """
+        is_invalid = None
+        aliases = self.current_file.get('Aliases', {})
+        if aliases and self.id_set_file:
+            aliased_fields = self.get_incident_fields_by_aliases(aliases)
+            aliases_with_invalid_marketplace = [alias.get("cliName") for alias in aliased_fields if self.is_alias_has_invalid_marketplaces(alias)]
+
+            if aliases_with_invalid_marketplace:
+                error_message, error_code = Errors.invalid_marketplaces_in_alias(aliases_with_invalid_marketplace)
+                is_invalid = self.handle_error(error_message, error_code, file_path=self.file_path,
+                                               warning=self.structure_validator.quite_bc)
+
+            aliases_with_inner_alias = [alias.get("cliName") for alias in aliased_fields if self.is_alias_has_inner_alias(alias)]
+            if aliases_with_inner_alias:
+                error_message, error_code = Errors.aliases_with_inner_alias(aliases_with_inner_alias)
+                is_invalid = self.handle_error(error_message, error_code, file_path=self.file_path,
+                                               warning=self.structure_validator.quite_bc)
+
+        return not is_invalid
+
+    def get_incident_fields_by_aliases(self, aliases: dict) -> List[dict]:
+        res = []
+        incident_fields_dict = self.get_incident_fileds_as_dict_from_id_set()
```

Comment:
I think we can skip the creation of that dict.
We already need to loop through the 'incidentFields' entry in the id set in order to find the details for the specific field (because its a list and not a dict).
We can extract the details while we are looping and return the json files from there, and then this method will be redundant.

Reply:
Very nice refactoring!!